File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        lastChar = part[-1]
        stack = []
        for char in s:
            stack.append(char)
            if(len(stack) >= len(part)) and (char == lastChar):
                if "".join(stack[-len(part):]) == part:
                    n = len(part)
                    while n>0:
                        stack.pop()
                        n-=1
        return "".join(stack)
        

File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.js
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
var removeOccurrences = function(s, part) {
    let stack = [], lastChar = part[part.length - 1]
    for(let char of s){
        stack.push(char)
        if(lastChar === char && stack.length >= part.length){
            if(stack.slice(-part.length).join("") === part){
                stack.length-= part.length
            }
        }
    }
    return stack.join("")
};